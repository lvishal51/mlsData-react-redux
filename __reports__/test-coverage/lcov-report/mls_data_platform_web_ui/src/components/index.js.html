<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
