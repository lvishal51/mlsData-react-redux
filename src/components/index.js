export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
