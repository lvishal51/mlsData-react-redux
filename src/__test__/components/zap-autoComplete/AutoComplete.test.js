import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { AutoCompleteComp as AutoComplete }  from 'components';

const functions = [
  {
    "name":"concat",
    "type":"fn(exprs_varargs: Column) -> Column",
    "doc":"Concatenates multiple input string columns together into a single string column.",
    "origin":"apache-spark-functions"
  },
  {
    "name":"concat_ws",
    "type":"fn(sep: string, exprs_varargs: Column) -> Column",
    "doc":"Concatenates multiple input string columns together into a single string column, using the given separator.",
    "origin":"apache-spark-functions"
  },
  {
    "name":"contains",
    "type":"fn(e: Column, other: Column) -> Column",
    "doc":"Contains the other column.",
    "origin":"apache-spark-functions"
  },
  {
    "name":"conv",
    "type":"fn(num: Column, fromBase: number, toBase: number) -> Column",
    "doc":"Convert a number in a string column from one base to another.",
    "origin":"apache-spark-functions"
  }
]

const inputProps = {
  MlsGetAutoCompleteOptionsData: {
    payload: {
      functions: functions
    }
  },
  MlsCanonicalActions: {
    getAutoCompleteOptions: jest.fn()
  },
  selectedFields: [{
    id: '7OtNhsQo9EL7OdyVImlf',
    tableSystemName: 'ADD0',
    standardName: 'null',
    shortName: 'Street Address',
    longName: 'Street Address',
    lookupName: 'Street Address',
    relevance: 100,
    active: false
  },
  {
    id: 'LfN3mCZe8QiN89tqVn8x',
    tableSystemName: 'OFFICE_14',
    standardName: 'null',
    shortName: 'Zip',
    longName: 'Zip',
    lookupName: 'null',
    relevance: 0,
    active: false
  }],
  value: "",
  udfs: [{"id":1,"enumValue":"SUM_ALL","description":"Sum of all numbers coming from raw columns data","active":true,"lastUpdateTs":1539791700000,"createTs":1539791700000,"expression":null}],
  onChange: jest.fn()
};

describe('Zap Autocomplete component', () => {
  it('should render correctly.', () => {
    const wraper = shallow(<AutoComplete {...inputProps}/>);
    expect(shallowToJson(wraper)).toMatchSnapshot();
  });

  it('handleInputChange function should be triggered correctly', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'con' } };
    const spy = jest.spyOn(wrapper.instance(), "handleInputChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDown function should be triggered correctly down', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    document.body.innerHTML = '<ul id="listBox" />';
    const event = { preventDefault() {}, keyCode: 40 };
    wrapper.setState({
      showAutoCompleteContainer: true
    });
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDown");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('keyDown', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDown function should be triggered correctly up', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    document.body.innerHTML = '<ul id="listBox" />';
    const event = { preventDefault() {}, keyCode: 38 };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDown");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('keyDown', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDown function should be triggered correctly esc', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    document.body.innerHTML = '<ul id="listBox" />';
    const event = { preventDefault() {}, keyCode: 27 };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDown");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('keyDown', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDown function should be triggered correctly esc', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    document.body.innerHTML = '<ul id="listBox" />';
    const event = { preventDefault() {}, keyCode: 32, ctrlKey: true };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDown");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('keyDown', event);
    expect(spy).toBeCalled();
  });

  it('handleKeyDown function should be triggered correctly esc', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    document.body.innerHTML = '<ul id="listBox" />';
    const event = { preventDefault() {}, keyCode: 67 };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDown");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('keyDown', event);
    expect(spy).toBeCalled();
  });

  it('complete function should be triggered correctly', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {} };
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "completeFunction");
    wrapper.find("li").at(0).simulate('click', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Down', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 40, target: { nextSibling: {focus: jest.fn() } }};
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Down', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 40, target: { parentNode: {firstChild: {focus: jest.fn() }}}};
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Up', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 38, target: { previousSibling: {focus: jest.fn() } }};
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Up', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 38, target: { parentNode:{ lastChild:{ focus: jest.fn()} }} };
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Esc', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 27, target: { parentNode: { classList: { remove: jest.fn() } } }};
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly Enter', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 13 };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    const spy2 = jest.spyOn(wrapper.instance(), "completeFunction");
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
    expect(spy2).toBeCalled();
  });

  it('handleKeyDownOption function should be triggered correctly for default', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const event = { preventDefault() {}, keyCode: 67 };
    const spy = jest.spyOn(wrapper.instance(), "handleKeyDownOption");
    document.body.innerHTML = '<input id="autocomplete" type="text"/>';
    wrapper.find("li").at(0).simulate('keyDown', event);
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });


  it('onFocusElement function should be triggered correctly', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onFocusElement");
    wrapper.find("li").at(0).simulate('focus');
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('onFocusAutoComplete function should be triggered correctly', () => {
    const wrapper = shallow(<AutoComplete {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onFocusAutoComplete");
    wrapper.setState({
      showAutoCompleteOptionInfo: true
    });
    wrapper.find("Input").at(0).simulate('focus');
    wrapper.instance().forceUpdate();
    expect(spy).toBeCalled();
  });

  it('componentWillUnmount should be called on unmount', () => {
    const component = shallow(<AutoComplete {...inputProps} />);
    const componentWillUnmount = jest.spyOn(component.instance(), 'componentWillUnmount');
    component.unmount();
    expect(componentWillUnmount).toHaveBeenCalled();
  });

  it('onClickOutsideAutoComplete function should be triggered correctly', () => {
    const map = {};
    const mockFn = jest.fn();
    document.addEventListener = jest.fn((event, mockFn) => {
      map[event] = mockFn;
    });
    const wrapper = mount(<AutoComplete {...inputProps}/>);
    map.click();
  });


});
