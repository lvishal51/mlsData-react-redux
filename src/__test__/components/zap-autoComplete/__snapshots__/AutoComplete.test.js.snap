// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Autocomplete component should render correctly. 1`] = `
<FormGroup
  className="position-relative"
  tag="div"
>
  <Input
    className="form-control"
    id="autocomplete"
    innerRef={[Function]}
    name="autocomplete"
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
  <ul
    className="autocomplete__container"
    id="listBox"
    role="listbox"
  >
    <li
      className="autocomplete__option variable"
      key="0"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="option"
      tabIndex="-1"
    >
      ADD0
    </li>
    <li
      className="autocomplete__option variable"
      key="1"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="option"
      tabIndex="-1"
    >
      OFFICE_14
    </li>
    <li
      className="autocomplete__option udf"
      key="2"
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      role="option"
      tabIndex="-1"
    >
      SUM_ALL
    </li>
  </ul>
  <div
    className="autocomplete__option__info"
  >
    <span
      className="autocomplete__option__info--type"
    />
    <span
      className="autocomplete__option__info--doc"
    />
  </div>
</FormGroup>
`;
